### ISSUE TITLE
Mapping a Bean to Map<String,Object> [Feature Request Or Documentation Update]

### ISSUE THREAD

**Issue by amaan75** [mapstruct/mapstruct#2504]
Can Mapstruct map a bean to a map<String,Object>
i am thinking along the lines of
Map<String,Object> convertToMap(SomePojo Bean);
all the fields become the string key, value is the value in the map but as an object :).
If this is doable with the current map struct then how about updating the reference guide and adding an example,
I can help with both.
- Our reference- and API [documentation](http://mapstruct.org/documentation/reference-guide/).
- Our [examples](https://github.com/mapstruct/mapstruct-examples) repository (contributions always welcome)

**Comment by filiphr** [comment]
Thanks for raising this issue @amaan75. In our upcoming 1.5 release we are going to support the reverse (`Map<String, Object>` to Bean).

I will leave this open to see if we can make it after the 1.5 beta. If you are interested in providing this functionality we are open in helping you out and reviewing a PR :)

**Comment by ereborDeveloper** [comment]
Yep! Looking for some opportunities for mapping Object to EntityDTO or Map<String, Object> to EntityDTO. Otherwise I should create Entity layer, but I don't want to do that because I already have postman collection with description of data response. It will be easy to use some "common" object :)

**Comment by amaan75** [comment]
@filiphr  sure, tell me how can i help, i'd be willing to raise a PR.

**Comment by vwiencek** [comment]
Hello, 

If I understand correctly right now `Map<String, String>` to bean is working, but not `Map<String, Object>` am I right ?

**Comment by filiphr** [comment]
@vwiencek please do not use issues to ask for questions. Mapping from `Map<String, ?>`  is working. You can read more about it in [Mapping Map to Bean](https://mapstruct.org/documentation/dev/reference/html/#mapping-map-to-bean) documentation

**Comment by tarvfx** [comment]
Hi, @filiphr 

> @vwiencek please do not use issues to ask for questions. Mapping from `Map<String, ?>` is working. You can read more about it in [Mapping Map to Bean](https://mapstruct.org/documentation/dev/reference/html/#mapping-map-to-bean) documentation

Inspired by you, I tried it, but the generated classes don't compile properly. Generated code is missing type casts! 

```java
@Mapper
public interface BeanMapper {

    BeanMapper INSTANCE = Mappers.getMapper(BeanMapper.class);

    Airfare toAirfare(Map<String, ?> map);
}
```

![image](https://github.com/mapstruct/mapstruct/assets/70388114/9d77e9b0-aa6e-40a7-b75e-a4f8fb5026cb)



BTW, one might prefer to use Map<String, Object> instead of Map<String, ?>. Thanks.

Best Regards,

**Comment by filiphr** [comment]
Sorry I was not clear enough. With `Map<String, ?>` I mean you to replace `?` to the type you want. If you use `Map<String, Object>` you'll get compiler errors (from MapStruct) saying that it cannot map from `Object` to `String` and etc.

Seems like you found a different bug with `Map<String, ?>`. You can create a separate issue for that so that we can look into it if you want to

**Comment by Arul-Pragasam-E1007898** [comment]
@filiphr do you have any update on this issue. Will this be supported by mapstruct anytime sooner?

**Comment by momonari8** [comment]
@filiphr any news on when this implementation will be available? Thanks!

**Comment by jason-waale** [comment]
@filiphr any news on when this will be available?

**Comment by filiphr** [comment]
@Arul-Pragasam-E1007898 , @momonari8 , @jason-waale  there is really no need to ping us. The MapStruct team is a team of volunteers that works on the project in their own free time. This issue is currently not planned for any release. If you would like to see it sooner then I would suggest that you try to contribute to MapStruct.

In the meantime we are focused on getting out a release for 1.6 and sorting out the remaining issues / PRs for that release

**Comment by ccarlenzoli** [comment]
@filiphr What about the process to join the team of volunteers?

**Comment by filiphr** [comment]
We don't have a special process. If you would like to contribute to MapStruct then you can do it. You can write comments in issues, do PRs, etc.

**Comment by davidricardo1026** [comment]
how about bean to map? still just map to bean

**Comment by amaan75** [comment]
hi @filiphr , the need for this has crept up in our code base once more, we resolved this with CGLib usage last time, but that is not supported with the newer JDKs, so I will try to pick this up soon.

**Comment by amaan75** [comment]
hi @filiphr, I have started working on this and i am in a bit of a pinch here,  i could make changes to MapMappingMethod by supplying some kind of a Bean base source RHS or i could do the same thing to BeanMappingMethod where the code for source is written but then i will have to make changes to supply the type Map as the output instead.
What do you suggest?

**Comment by filiphr** [comment]
Thanks for working on this @amaan75. I think that the `BeanMappingMethod` would indeed fit a bit better for this. Or maybe even some new `BeanToMapMappingMethod`, but not sure how much worth that is, there might be a lot of things that you need from the `BeanMappingMethod`

**Comment by zheng-kai** [comment]
@amaan75 Is this feature available now? I need it for my work too. If it's not ready yet, I'd be happy to help out.

**Comment by xiananliu** [comment]
Is this feature available now? So many years have passed.  thx
